geode-core/src/main/java/org/apache/geode/internal/admin/SSLConfig.java

```diff
@@ -237,10 +249,26 @@ public void toDSProperties(Properties props) {
     public Builder() {}
 
     public SSLConfig build() {
+
+      final String sniProxyProperty = System.getProperty(GEMFIRE_PREFIX + "security.sni-proxy");
```

Comment:
Yeah, it's currently a "security" property merely because SNI is part of TLS and TLS is configured via "security" properties.

The URL idea is interesting! I'm letting it soak in now…

Reply:
this will be addressed in a follow-on story [GEODE-7852](https://issues.apache.org/jira/browse/GEODE-7852)